src/GafferOSL/OSLObject.cpp

```diff
@@ -254,7 +275,7 @@ IECore::ConstObjectPtr OSLObject::computeProcessedObject( const ScenePath &path,
 	PrimitiveVariable::Interpolation interpolation = static_cast<PrimitiveVariable::Interpolation>( interpolationPlug()->getValue() );
 
 	IECoreScene::ConstPrimitivePtr resampledObject = IECore::runTimeCast<const IECoreScene::Primitive>( resampledInPlug()->objectPlug()->getValue() );
-	CompoundDataPtr shadingPoints = prepareShadingPoints( resampledObject.get() );
+	CompoundDataPtr shadingPoints = prepareShadingPoints( resampledObject.get(), resampledNamesPlug()->getValue() );
```

Comment:
If a plug is pulled on in `compute()`, it should also be hashed in `hash()`. I suspect we can get away with this here because it is hashed into `resampledInPlug()`, but it's best to stick to the rules - it's easy to get things subtly wrong otherwise. The same goes for `affects()`.